app/src/androidTest/java/com/example/bigowlapp/activity/SignUpPageActivityTest.java

```diff
@@ -0,0 +1,202 @@
+    private String email = "john.doe@email.com";
+    private String password = "abc123";
+    private String phone;
+
+    SignUpPageActivity signUpPageActivity;
+
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void unSufficientSignUpInputTest() {
+        onView(withId(R.id.button_sign_up))
+                .perform(scrollTo())
+                .check(matches(withText("Sign Up"))).perform(click());
+        onView(withId(R.id.edit_text_phone)).check(matches(hasErrorText("The string supplied did not seem to be a phone number.")));
+    }
+
+    @Test
+    public void unValidSignUpInputTest() {
+        onView(withId(R.id.user_first_name)).perform(typeText(firstName), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.user_last_name)).perform(typeText(lastName), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.edit_text_text_mail_address)).perform(typeText(email), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.edit_text_text_password)).perform(typeText(password), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.edit_text_phone)).perform(typeText("5141234567"), ViewActions.closeSoftKeyboard());
+
+        onView(withId(R.id.button_sign_up))
+                .perform(scrollTo())
+                .check(matches(withText("Sign Up"))).perform(click());
+
+        Espresso.onView(isRoot()).perform(waitFor(1000));
+        Espresso.onView(withId(R.id.user_first_name)).check(matches(isDisplayed()));
+        Espresso.onView(withId(R.id.user_last_name)).check(matches(isDisplayed()));
+        Espresso.onView(withId(R.id.edit_text_text_mail_address)).check(matches(isDisplayed()));
+        Espresso.onView(withId(R.id.edit_text_text_password)).check(matches(isDisplayed()));
+        Espresso.onView(withId(R.id.edit_text_phone)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void validSignUpInputTest() {
+        //create randomness for phone number, test purpose only.
+        Random rand = new Random();
+        int min = 0;
+        int max = Integer.MAX_VALUE;
+        int result = rand.nextInt(max - min) + min;
+
+        this.phone = String.valueOf(result);
+        this.email = result + "@email.com";
+
+        onView(withId(R.id.user_first_name)).perform(typeText(firstName), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.user_last_name)).perform(typeText(lastName), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.edit_text_text_mail_address)).perform(typeText(this.email), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.edit_text_text_password)).perform(typeText(password), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.edit_text_phone)).perform(typeText(this.phone), ViewActions.closeSoftKeyboard());
+
+        onView(withId(R.id.button_sign_up))
+                .perform(scrollTo())
+                .check(matches(withText("Sign Up"))).perform(click());
+    }
```

Comment:
This is a good test, but the issue is it pops in dummy users to the real database. SignUpViewModel should be mocked to prevent this from happening, and upon doing that, I verify can be added to confirm the creation of user is actually completed.

You can check out some of the other tests to see how that done, since we have similar ideas through out where the ViewModel of an activity/fragment is mocked, and instead of actually allowing it to touch the db, we instead verify calls onto the view models.